### PR TITLE
Add Deprecation Process doc

### PR DESCRIPTION
**Pull request by dsouzai**
This doc reflects the discussion between @DanHeidinga @pshipton @keithc-ca @0xdaryl and myself regarding the Eclipse OpenJ9 Deprecation Process.

### CODE REVIEW

**Comment by dsouzai** [comment]
I'd appreciate all reviews; I tried to make it as concise as possible but parts of it still feel a little clumsy :).

File: doc/processes/deprecation.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+Java version that precedes the version in which the deprecation takes
+effect.
+* When the codebase is updated to reflect the deprecation, it will need
+to be `#ifdef`'d so that previous Java versions continue to have the old
+behaviour. As such, any relevant documentation that needs to be updated
+to reflect the deprecation needs to do so by removing the text that
+warns about the upcoming changes, replacing it with a message stating
+that the behaviour is deprecated in version n, and including new text
+describing the new behaviour from version n onward.
+* If the deprecation involves a flip in behaviour that is controlled
```
Comment (DanHeidinga) [inline doc/processes/deprecation.md:54]: This is true but does it need to be part of the process?

File: doc/processes/deprecation.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+to reflect the deprecation.
+4. Remove any relevant documentation once all versions prior to the
+version n go out of service.
+
+## Important Notes
+
+* Deprecation can only occur when a new Java version is released; this
+should not be confused with a new OpenJ9 version, which may not
+necessarily include a new Java version.
+* Since an OpenJ9 release can include multiple Java versions, the
```
[the remark this answers is not in the document]
Comment (dsouzai) [reply doc/processes/deprecation.md:43]: That's a good point, I suppose if OpenJ9 version X does not contain a new Java version (which I don't know how often it'll really happen), but version Y and Z will, there's no harm in having the deprecation notice added to X and then actually deprecated in Z. I suppose as long as we also remember to add the notice to the release notes of Y (in addition to X) then it wouldn't contradict anything else in the rest of the doc.

File: doc/processes/deprecation.md (hunk 1 above)
Comment (dsouzai) [reply doc/processes/deprecation.md:54]: No, I don't think it needs to. I had arranged this doc to have the actual policy in the `Eclipse OpenJ9 Deprecation Process` section, and notes or clarifications in the `Important Notes` section. However, if the idea is to view the doc in its entirety as the policy, then I can remove this bullet as well.

Comment (DanHeidinga) [reply doc/processes/deprecation.md:54]: I think we're better to keep the policy document short and simple.  If we need several notes of clarification, then maybe we need to re-examine the policy.

**Comment by dsouzai** [comment]
@DanHeidinga made the requested changes; could you review again.

**Review by DanHeidinga — APPROVED** [review]
Works for me.  Will merge in a few days unless @pshipton or @mstoodle (or others) chime in with concerns

**Review by pshipton — APPROVED** [review]

**Pull request merged**